app/setting.go

```diff
@@ -2,9 +2,33 @@ package app
 
 import (
 	"github.com/pborman/uuid"
+	metadata "github.com/rancher/kontainer-driver-metadata/rke"
+	kd "github.com/rancher/rancher/pkg/controllers/management/kontainerdrivermetadata"
 	"github.com/rancher/rancher/pkg/settings"
 )
 
 func addSetting() error {
-	return settings.InstallUUID.SetIfUnset(uuid.NewRandom().String())
+	if err := settings.InstallUUID.SetIfUnset(uuid.NewRandom().String()); err != nil {
+		return err
+	}
+	return addK8sVersionData()
+}
+
+func addK8sVersionData() error {
+	rancherVersion := settings.ServerVersion.Get()
+	if rancherVersion == "dev" || rancherVersion == "master" {
+		rancherVersion = kd.RancherVersionDev
+	}
+	k8sCurrVersions := map[string]interface{}{}
+
+	k8sVersionToRKESystemImages, _, k8sVersionToSvcOptions := kd.GetK8sVersionInfo(rancherVersion,
+		metadata.DriverData.K8sVersionRKESystemImages,
+		metadata.DriverData.K8sVersionWindowsSystemImages,
+		metadata.DriverData.K8sVersionServiceOptions,
+		metadata.DriverData.K8sVersionInfo)
```

Comment:
we seem to be doing this during server start twice, (main.go and addSetting) - maybe we could cache the results of kd.GetK8sVersionInfo for the current rancherVersion?

Reply:
main.go is `./pkg/image/export/main.go`, which is called only when building dapper, and it prints out the images ... but not used in actual run